docs/get-started/authenticate-without-flask-strava.md

```diff
@@ -0,0 +1,414 @@
+```
+
+### Time to put the above into action 
+
+Above you created your Strava authentication url that will allow a user to 
+gives you (read-only) permission to access their data on Strava via the API. 
+
+Below you will use that URL using the Python 
+`webbrowser.open` method. This will allow you to:
+
+1. Open the url in a web browser directly from your Python code, 
+2. Authenticate with Strava (this step you will login to Strava and provide permission for your code to access the developer app that you created above).
+3. If you are working locally and not building an app, once you have provided access to Strava, you will get a `page not found` page with a long URL. It looks like a mistake, but really the url in your browser contains the code that you need to authenticate with the API! Don't close the browser window just yet.
+
+
+`http://127.0.0.1:5000/authorization?state=&code=xxxxxxxxxxx&scope=read`.
+
+4. Copy the values in the URL between **code** and the **&** symbol. Above, the 
+x's represent the values that you need to copy.
+
+```python
+# Open the url that you created above in a web browser 
+webbrowser.open(url)
+    print(
+        """You will see a url that looks like this. """,
+        """http://127.0.0.1:5000/authorization?state=&code=12323423423423423423423550&scope=read,activity:read_all,profile:read_all,read_all")""",
+        """Copy the values between code= and & in the url that you see in hte 
+        browser. """,
+    )
+    # Using input allows you to copy the code into your Python console 
+    # (or Jupyter Notebook)
+    code = input("Please enter the code that you received: ")
+    print(
+        "Great! Your code is ",
+        code,
+        "Next I will exchange that code for a token.\n"
+        "I only have to do this once.",
+    )
+```
+
+You only need to get a code from Strava once. 
+
+Next you are ready to exchange the code for a token. After this, you can 
+refresh your token as often as you need to to grab your data. 
+
+```python
+token_response = client.exchange_code_for_token(
+        client_id=client_id, client_secret=client_secret, code=code
+    )
+# Save token locally as a pickle 
+with open(path_to_save, "wb") as f:
+    pickle.dump(token_response, f)
+
+print("Token saved - hooray!")
+
+# Access token and refresh tokens
+access_token = token_response['access_token']
+refresh_token = token_response['refresh_token']  # You'll need this in 6 hours
+```
+
+## Token refresh 
+
+The access token that you are provided above is good for 6 hours. After 6 
+hours it expires. However you can refresh it as needed if you wish to grab 
+more data in the future. 
+
+Above you save the token as a pickle so you can access it in the future. 
+
+Once you have the access token, you can begin to 
+
+```python
+# TODO: Is it ok to assign the attribute this way? Shouldn't it get assigned when 
+# you use the code / token exchange method? 
```

Comment:
This looks like a great idea to improve the client API, i.e., let the `exchange_code_for_token()` method set the retrieved access token on itself as a side-effect.

Reply:
I believe that this happens now. i'll resolve but from what i can see we no longer need to manually reset it.